frontend/src/components/HeaderNav.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { Component, useState, useRef, useEffect } from 'react';
+import { StyleSheet, View, BackHandler } from 'react-native';
+import { Appbar, Searchbar } from 'react-native-paper';
+import { connect, useDispatch } from 'react-redux';
+import StoreSelector from './StoreSelectorButton';
+import { HeaderStates, GlobalStates } from '../constants/States';
+import { State } from '../constants/Interfaces';
+import { ActionTypes } from '../constants/ActionTypes';
+import { querySuggestions } from './Helper/AsyncCalls';
+
+interface Props {
+    currState: HeaderStates
+}
+
+// header navigation for app
+function HeaderNav(props: Props) {
+    const dispatch = useDispatch();
+    let [clearQuery, setClearQuery] = useState(() => false);
```

Comment:
Why is this a function that evaluates to false rather than just being `false`?

Reply:
Same thing with `query` and `hasBackEvent`

Reply:
Okay, so I read the documentation, and the way that you did it is technically legal, but it's really meant as an optimization for cases when the initial state is expensive to compute. I think it would be easier to read if we dropped the function